### PR TITLE
Add 1es-redirect template. Support template validation canary runs

### PR DESCRIPTION
**Pull request by benbp**

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -82,40 +75,18 @@ parameters:
...
-      skipBuildTagsForGitHubPullRequests: true
-    sdl:
-      sourceAnalysisPool:
-        name: azsdk-pool-mms-win-2022-general
-        image: azsdk-pool-mms-win-2022-1espt
-        os: windows
-      eslint:
-        enabled: false
-        justificationForDisabling: "ESLint injected task has failures because it uses an old version of mkdirp. We should not fail for tools not controlled by the repo. See: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3556850"
-      codeql:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml:102]: Are these not needed? As I don't see it 1es-redirect?

Comment (benbp) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:102]: Oh might have been a mismatch/gap, thanks for catching.

Comment (benbp) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:102]: Unless @scbedd when you added the initial 1es-redirect you intended to remove these?

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:102]: ESLint and codeql are definitely necessary. 

Git history says I was the one to add it, but I definitely don't recall using 1es redirect for any of the python yml. No idea how that happened.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**